tsdb/engine/wal/wal.go

```diff
@@ -0,0 +1,1591 @@
+/*
+Package WAL implements a write ahead log optimized for write throughput
+that can be put in front of the database index.
+
+The WAL is broken into different partitions. The default number of
+partitions is 5. Each partition consists of a number of segment files.
+By default these files will get up to 2MB in size before a new segment
+file is opened. The files are numbered and start at 1. The number
+indicates the order in which the files should be read on startup to
+ensure data is recovered in the same order it was written.
+
+Partitions are flushed and compacted individually. One of the goals with
+having multiple partitions was to be able to flush only a portion of the
+WAL at a time.
+
+The WAL does not flush everything in a partition when it comes time. It will
```

Comment:
I want to be sure I understand why this is done this way. If everything in the partition was flushed, it might be the case that there are many small amounts of data for many series. So flushing the entire thing would result in poor performance into Bolt. To get decent performance, we must flush a decent amount of data for each series, when the flush comes. Is that correct?